shm/src/player.cpp

```diff
@@ -36,5 +36,9 @@ void Player::countAvailableSpace() {
 }
 
 void Player::payCrew(size_t money) {
-    money_ -= money;
+    if (money_ < money) {
+        money_ = -1;
+    } else {
+        money_ -= money;
+    }
```

Comment:
You can't set money_ to -1 because it's size_t.
```suggestion
    (money_ < money) ? money_ = 0 : money_ -= money;
```

Reply:
Yes, but we can't set the money to 0 as it can still be viable value. I think I will set it to maximum size_t value?